spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala

```diff
@@ -36,8 +36,13 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
  */
 @SerialVersionUID(- 8525406230282608924L)
-class Input[T: ClassTag]()(implicit ev: TensorNumeric[T])
+class Input[T: ClassTag](inputShape: Array[Int])(implicit ev: TensorNumeric[T])
```

Comment:
Should we define a Shape class here? It will make the code easy to understand and extend.

Reply:
BTW, how to represent a not fixed size, e.g. the batch size dimension.

Reply:
The inputShape which specified by user would not containing batch same as Keras.